tests/models/hf/test_hf_config.py

```diff
@@ -235,3 +237,44 @@ def test_nested_override():
     assert isinstance(model.config.ffn_config, PretrainedConfig)
     # Ensure the other values still exist and are not set back to their defaults
     assert model.config.ffn_config.moe_num_experts == 16
+
+
+@pytest.mark.gpu
+def test_use_flash():
+    model_cfg = {
+        'name': 'hf_causal_lm',
+        'pretrained_model_name_or_path': 'codellama/CodeLlama-7b-hf',
+        'config_overrides': {
+            'num_hidden_layers': 2,
+            'hidden_size': 32,
+            'intermediate_size': 64,
+            'torch_dtype': 'bfloat16',
```

Comment:
is it necessary that the `torch_dtype` in the config is set to bf16?

I thought we would want to test that the model loaded in fp32 is able to use flash attn?

Reply:
ah yeah, i snuck into this test that hf doesn't silently put the master weights into the specified dtype :P

Reply:
so that's why you have the check that the params are in fp32 then?

can you add a comment clarifying that :)

Reply:
done